drgn_tools/task.py

```diff
@@ -197,6 +197,91 @@ def get_command(task: Object) -> str:
     return escape_ascii_string(task.comm.string_())
 
 
+def get_ppid(task: Object) -> int:
+    """
+    :returns: Parent PID of the task
+    """
+    return task.parent.pid.value_()
+
+
+def get_rss(prog: Program, task: Object) -> int:
+    """
+    :returns RSS value
+    """
+    try:
+        filerss = task.mm.rss_stat.count[0].counter.value_()
+        if filerss < 0:
+            filerss = 0
+        anonrss = task.mm.rss_stat.count[1].counter.value_()
+        if anonrss < 0:
+            anonrss = 0
+        shmrss = task.mm.rss_stat.count[3].counter.value_()
+        if shmrss < 0:
+            shmrss = 0
+        rss_mm = filerss + anonrss
+
+    except Exception:
+        rss_mm = 0
+
+    rss_task = 0
+    gid = task.tgid.value_()
+    for gtask in for_each_task(prog):
+        if gid == gtask.tgid.value_():
+            task_filerss = task_anonrss = task_shmrss = 0
+
+            try:
+                task_filerss = gtask.rss_stat.count[0].value_()
+                if task_filerss < 0:
+                    task_filerss = 0
+
+                task_anonrss = gtask.rss_stat.count[1].value_()
+                if task_anonrss < 0:
+                    task_anonrss = 0
+
+                task_shmrss = gtask.rss_stat.count[3].value_()
+                if task_shmrss < 0:
+                    task_shmrss = 0
+
+                rss_task += task_filerss + task_anonrss
+
+            except Exception:
+                rss_task = 0
+
+    rss = rss_mm + rss_task
+    page_size = prog["PAGE_SIZE"].value_()
+    rss_kb = (rss * page_size) // 1024
+    return rss_kb
+
+
+def get_pct_mem(prog: Program, task: Object) -> str:
+    """
+    : returns %MEM value
+    """
+    try:
+        rss = get_rss(prog, task)
+        page_shift = prog["PAGE_SHIFT"].value_()
+        page_offset_base = prog["page_offset_base"].value_()
+        high_mem = prog["high_memory"].value_()
+        total_pages = ((high_mem - page_offset_base) >> page_shift) * 4
+        mem = (rss * 100) / total_pages
+        pct_mem = "%.1f" % mem
+    except Exception:
+        pct_mem = str(0.0)
+    return pct_mem
```

Comment:
Again, what's the `except Exception` for? It needs to have a good documented reason, or a more specific exception with a reason.

And `str(0.0)` can just be written as `"0.0"`, a bit easier that way :)

Reply:
Don't really need any check for errors here. RSS calculation part of the code will take care of it so I have removed that.